### ISSUE TITLE
Fix memory leak

### ISSUE THREAD

**Issue by uli42** [ArcticaProject/nx-libs#541]
```
commit 5e106a71b9f8077216d41619402952b0005dd8a4
Author: Adam Jackson <ajax@nwnk.net>
Date:   Wed Mar 15 16:49:04 2006 +0000

    Coverity #807: Fix a memory leak in XFixesExpandRegion.

diff --git a/xfixes/region.c b/xfixes/region.c
index d317f73..68c7015 100755
--- a/xfixes/region.c
+++ b/xfixes/region.c
@@ -839,6 +839,7 @@ ProcXFixesExpandRegion (ClientPtr client)
            REGION_INIT (pScreen, &r, &pTmp[i], 0);
            REGION_UNION (pScreen, pDestination, pDestination, &r);
        }
+       xfree(pTmp);
     }
     if (ret == Success) 
        ret = client->noClientException;
```

**Comment by frantracer** [comment]
Hello,
I was running the latest version so far (in this moment is commit 7f777e01e3bb8780ba91e70171a0aae78a95692e) and I found a memory leak in the nx-agent during the execution of any kind of application that streams data, like spotify or firefox with youtube. After half an hour aproximately the kernel had to kill it because it was using almost the 100% of the memory.

Do you know whether the problem I described is related to this issue?

**Comment by uli42** [comment]
Probably not. But I encourage you to test it!

Vom Smartphone gesendet.

----- Ursprüngliche Nachricht -----
Von: "frantracer" <notifications@github.com>
Gesendet: ‎29.‎11.‎2017 17:25
An: "ArcticaProject/nx-libs" <nx-libs@noreply.github.com>
Cc: "Ulrich Sibiller" <uli42@gmx.de>; "Author" <author@noreply.github.com>
Betreff: Re: [ArcticaProject/nx-libs] Fix memory leak (#541)

Hello,
I was running the latest version so far (in this moment is commit 7f777e0) and I found a memory leak in the nx-agent during the execution of any kind of application that streams data, like spotify or firefox with youtube. After half an hour aproximately the kernel had to kill it because it was using almost the 100% of the memory.
Do you know whether the problem I described is related to this issue?

**Comment by uli42** [comment]
Thinking again: Some of the leaks are in the logging code, the longer it runs the more it logs. So it could help. Please test!

Vom Smartphone gesendet.

----- Ursprüngliche Nachricht -----
Von: "Ulrich Sibiller" <ulrich.sibiller@gmail.com>
Gesendet: ‎29.‎11.‎2017 18:33
An: "ArcticaProject/nx-libs" <reply@reply.github.com>; "ArcticaProject/nx-libs" <nx-libs@noreply.github.com>
Cc: "Ulrich Sibiller" <uli42@gmx.de>; "Author" <author@noreply.github.com>
Betreff: AW: [ArcticaProject/nx-libs] Fix memory leak (#541)

Probably not. But I encourage you to test it!

Vom Smartphone gesendet.


Von: frantracer
Gesendet: ‎29.‎11.‎2017 17:25
An: ArcticaProject/nx-libs
Cc: Ulrich Sibiller; Author
Betreff: Re: [ArcticaProject/nx-libs] Fix memory leak (#541)


Hello,
I was running the latest version so far (in this moment is commit 7f777e0) and I found a memory leak in the nx-agent during the execution of any kind of application that streams data, like spotify or firefox with youtube. After half an hour aproximately the kernel had to kill it because it was using almost the 100% of the memory.
Do you know whether the problem I described is related to this issue?
